GVFS/GVFS.Common/Git/GitObjects.cs

```diff
@@ -47,9 +47,10 @@ public virtual bool TryDownloadCommit(string commitSha)
             const bool PreferLooseObjects = false;
             IEnumerable<string> objectIds = new[] { commitSha };
 
+            GitProcess gitProcess = new GitProcess(this.Enlistment);

```

Comment:
Would this be better as a member variable instead of passing it around everywhere?

Reply:
We need to pass it around to allow us to stop the background prefetch during an unmount (the `PrefetchStep` gets a `GitProcess` from its base class, and the base class handles shutting down the process on `Stop()`). Other commands that use those methods need to create their own process to pass to those methods.

Reply:
I added a GitProcess to GVFSEnlistment in a previous PR.  Wondering if this should have been done at Enlistment itself.  Then this class could just take advantage of it.